src/timesync/timesyncd-manager.c

```diff
@@ -524,10 +524,21 @@ static int manager_receive_response(sd_event_source *source, int fd, uint32_t re
         receive = ntp_ts_to_d(&ntpmsg.recv_time);
         trans = ntp_ts_to_d(&ntpmsg.trans_time);
         dest = ts_to_d(recv_time) + OFFSET_1900_1970;
+        precision = exp2(ntpmsg.precision);
 
         offset = ((receive - origin) + (trans - dest)) / 2;
         delay = (dest - origin) - (trans - receive);
 
+        /* Some cheap routers have a precision of only 1 second and round recv_time to it
+         * Any time difference less than the precision probably due to rounding so subtract it.
+         */
+        if (offset < -precision)
+                offset += precision;
+        else if (offset > precision)
+                offset -= precision;
+        else
+                offset = 0;
```

Comment:
Hm? Why `0` is assigned?

Reply:
I have added the offset_adjusted as suggested

Reply:
I have added comments on each of the three cases with worked examples of how the broken rounding on the server affects the responses, hopefully they should clear up why offests less than the precision should be ignored.

In the process I realises that `[-precision ... precision]` is actually 2×precision wide where as the rounding error is at worst 1×precision, so given that the noncompliant routers in question seem to round down, I adjusted the logic to match.

The comments are quite verbose so if you think they should be removed let me know.

Reply:
I also added a debug line for each case.